controllers/argocd/notifications/deployment.go

```diff
@@ -0,0 +1,212 @@
+package notifications
+
+import (
+	"time"
+
+	"github.com/argoproj-labs/argocd-operator/common"
+	"github.com/argoproj-labs/argocd-operator/pkg/argoutil"
+	"github.com/argoproj-labs/argocd-operator/pkg/cluster"
+	"github.com/argoproj-labs/argocd-operator/pkg/mutation"
+	"github.com/argoproj-labs/argocd-operator/pkg/workloads"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func (nr *NotificationsReconciler) reconcileDeployment() error {
+
+	nr.Logger.Info("reconciling deployments")
+
+	name := argoutil.GenerateUniqueResourceName(nr.Instance.Name, nr.Instance.Namespace, ArgoCDNotificationsControllerComponent)
+
+	notificationEnv := nr.Instance.Spec.Notifications.Env
+	notificationEnv = argoutil.EnvMerge(notificationEnv, ProxyEnvVars(), false)
+
+	podSpec := corev1.PodSpec{
+		Volumes: []corev1.Volume{
+			{
+				Name: TLSCerts,
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: common.ArgoCDTLSCertsConfigMapName,
+						},
+					},
+				},
+			},
+			{
+				Name: ArgoCDRepoServerTLS,
+				VolumeSource: corev1.VolumeSource{
+					Secret: &corev1.SecretVolumeSource{
+						SecretName: common.ArgoCDRepoServerTLSSecretName,
+						Optional:   argoutil.BoolPtr(true),
+					},
+				},
+			},
+		},
+		Containers: []corev1.Container{{
+			Command:         GetNotificationsCommand(nr.Instance),
+			Image:           GetArgoContainerImage(nr.Instance),
+			ImagePullPolicy: corev1.PullAlways,
+			Name:            common.ArgoCDNotificationsControllerComponent,
+			Env:             notificationEnv,
+			Resources:       GetNotificationsResources(nr.Instance),
+			LivenessProbe: &corev1.Probe{
+				ProbeHandler: corev1.ProbeHandler{
+					TCPSocket: &corev1.TCPSocketAction{
+						Port: intstr.IntOrString{
+							IntVal: int32(9001),
+						},
+					},
+				},
+			},
+			SecurityContext: &corev1.SecurityContext{
+				AllowPrivilegeEscalation: argoutil.BoolPtr(false),
+				Capabilities: &corev1.Capabilities{
+					Drop: []corev1.Capability{
+						CapabilityDropAll,
+					},
+				},
+			},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      TLSCerts,
+					MountPath: VolumeMountPathTLS,
+				},
+				{
+					Name:      ArgoCDRepoServerTLS,
+					MountPath: VolumeMountPathRepoServerTLS,
+				},
+			},
+			WorkingDir: WorkingDirApp,
+		}},
+		SecurityContext: &corev1.PodSecurityContext{
+			RunAsNonRoot: argoutil.BoolPtr(true),
+		},
+	}
+
+	AddSeccompProfileForOpenShift(nr.Logger, nr.Client, &podSpec)
```

Comment:
I think this function should be a mutation, and registered with the other mutations separately, so it need not be called here 

I will include that in a separate PR. Please remove this line

Reply:
@ciiay please remove this line from here